core/src/main/java/org/infinispan/commands/functional/TxReadOnlyManyCommand.java

```diff
@@ -107,6 +106,7 @@ public Object perform(InvocationContext ctx) throws Throwable {
             EntryView.ReadWriteEntryView rw = EntryViews.readWrite(entry);
             for (Mutation<K, V, ?> mutation : mutations) {
                ret = mutation.apply(rw);
+               entry.updatePreviousValue();
```

Comment:
During command execution with RR entries, the 'actual value' is taken from the `oldValue` - see `RepeatableReadEntry.setValue()`, used e.g. in `ReplaceCommand.perform()` or `PKVC.performPut:271`. Some places don't use that, e.g. the functional commands, but these don't deal with retries properly anyway. Actually `RemoveCommand` does not use it either - this could be a bug.
So, we have to finalize the application of mutation by calling `updatePreviousValue()` in order to  get 'correct' previous value returned by `setValue()`in next mutation. This is usually done by EWI after each command, but here we have to call it manually, since the mutations don't go through the stack.